templates/contribute/style.md

```diff
@@ -107,6 +109,22 @@ theorem nat_case {P : nat → Prop} (n : nat) (H1: P 0) (H2 : ∀m, P (succ m))
 nat.induction_on n H1 (assume m IH, H2 m)
 ```
 
+If the theorem statement requires multiple lines, indent the subsequent lines:
+```lean
+namespace nat
+
+lemma le_induction {P : ℕ → Prop} {m}
+  (h0 : P m) (h1 : ∀ n, m ≤ n → P n → P (n + 1)) :
+  ∀ n, m ≤ n → P n :=
+by apply nat.less_than_or_equal.rec h0; exact h1
+
+def decreasing_induction {P : ℕ → Sort*} (h : ∀ n, P (n + 1) → P n) {m n : ℕ} (mn : m ≤ n)
+  (hP : P n) : P m :=
+le_rec_on mn (λ k ih hsk, ih $ h k hsk) (λ h, h) hP
```

Comment:
Playing devils advocate - is this more readable as
```suggestion
lemma le_induction {P : ℕ → Prop} {m}
(h0 : P m) (h1 : ∀ n, m ≤ n → P n → P (n + 1)) :
  ∀ n, m ≤ n → P n :=
by apply nat.less_than_or_equal.rec h0; exact h1

def decreasing_induction {P : ℕ → Sort*} (h : ∀ n, P (n + 1) → P n) {m n : ℕ} (mn : m ≤ n)
(hP : P n) :
  P m :=
le_rec_on mn (λ k ih hsk, ih $ h k hsk) (λ h, h) hP
```
or
```suggestion
lemma le_induction {P : ℕ → Prop} {m}
  (h0 : P m) (h1 : ∀ n, m ≤ n → P n → P (n + 1)) :
    ∀ n, m ≤ n → P n :=
by apply nat.less_than_or_equal.rec h0; exact h1

def decreasing_induction {P : ℕ → Sort*} (h : ∀ n, P (n + 1) → P n) {m n : ℕ} (mn : m ≤ n)
  (hP : P n) :
    P m :=
le_rec_on mn (λ k ih hsk, ih $ h k hsk) (λ h, h) hP
```
Essentially, do we want some subset of the addition rules:
* Wrapped arguments are not indented
* A newline must appear after the colon if the argument list spanned multiple lines
* If a newline appears after a colon, the theorem statement gains another level of indentation

Reply:
Note that the goal of the pull request is descriptive: I wrote down what I understand as the current mathlib style. If we want to make changes to the conventions, let's make a poll on Zulip first.

For the record: I agree that it's almost always more readable to require a line break after `:` and `:=`, with the main exception being when the theorem is called  `bar_of_foo`, so we'd expect the conclusion line to be `foo -> bar`. If we turn `foo` into a named argument, I prefer keeping it on the same line. Something like:
```lean
lemma no_divisors_of_gcd_eq_one {x y : R}
  (h : gcd x y = 1) : ∀ z, z | x -> z | y -> is_unit z :=
sorry
```